Hadrons/Modules/MNPR/Bilinear.hpp

```diff
@@ -145,42 +145,49 @@ void TBilinear<FImpl>::execute(void)
     // Propagators
     auto  &qIn    = envGet(PropagatorField, par().qIn);
     auto  &qOut   = envGet(PropagatorField, par().qOut);
+    LatticeSpinColourMatrix qIn_phased(env().getGrid()), qOut_phased(env().getGrid());
     envGetTmp(ComplexField, pDotXIn);
     envGetTmp(ComplexField, pDotXOut);
     envGetTmp(ComplexField, xMu);
 
     // momentum on legs
     //TODO: Do we want to check the momentum input format? Not done in MSink::Point, so probably ok like this.
     std::vector<Real>           pIn  = strToVec<Real>(par().pIn), 
-	                        pOut = strToVec<Real>(par().pOut);
+	                            pOut = strToVec<Real>(par().pOut); // Should this be Int instead of Real?
```

Comment:
We should think about validating the input anyway - forcing it to be int is one way, but also we want to enforce exactly 4 numbers etc. This is not done consistently in Hadrons. I'll raise the question with Antonin whether we want a routine which valdates all four-momentum inputs in Hadrons.